cylc/flow/async_util.py

```diff
@@ -389,7 +389,7 @@ def _scandir(future, path, request):
     """Callback helper for scandir()."""
     future.set_result([
         Path(path, directory.name)
-        for directory in request.result
+        for directory in request.result or []
```

Comment:
Why does this make any difference?  If I put an empty directory in `~cylc-run` then `request.result` is an empty list already.  (And I can seem to

Reply:
I saw an error where (`request.result` was returning `None`) in the log whilst working on cylc-uiserver so put this in to suppress it.

Perhaps it happens when you scan a non-existent directory or something like that.

Reply:
OK, fair enough.  Probably needs a comment then, because it looks weird (if you expect `request.result` to always be list).

Reply:
Huh, GitHub let me create a "suggestion" and commit it to the branch :grin:

Reply:
One review will do for this.

Reply:
Brilliant, my comment was a style violation (line too long) :rofl:

Reply:
Hot fix coming.